apiserver/migrationtarget/migrationtarget_test.go

```diff
@@ -256,19 +257,144 @@ func (s *Suite) TestAdoptResources(c *gc.C) {
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(results.Results, gc.HasLen, 1)
+	c.Assert(results.Results[0].Error, gc.ErrorMatches, `couldn't find instance "birds" for machine 1`)
+}
+
+func (s *Suite) TestCheckMachinesExtraInstances(c *gc.C) {
+	st := s.Factory.MakeModel(c, nil)
+	defer st.Close()
+
+	fact := factory.NewFactory(st)
+	fact.MakeMachine(c, &factory.MachineParams{
+		InstanceId: "judith",
+	})
+	env := mockEnviron{
+		Stub: &testing.Stub{},
+		instances: []*mockInstance{
+			{id: "judith"},
+			{id: "analyse"},
+		},
+	}
+	api := s.mustNewAPIWithEnviron(c, &env)
+
+	results, err := api.CheckMachines(
+		params.ModelArgs{ModelTag: st.ModelTag().String()})
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(results.Results, gc.HasLen, 1)
+	c.Assert(results.Results[0].Error, gc.ErrorMatches, `no machine with instance "analyse"`)
+}
+
+func (s *Suite) TestCheckMachinesErrorGettingInstances(c *gc.C) {
+	st := s.Factory.MakeModel(c, nil)
+	defer st.Close()
+
+	env := mockEnviron{Stub: &testing.Stub{}}
+	env.SetErrors(errors.Errorf("kablooie"))
+	api := s.mustNewAPIWithEnviron(c, &env)
+
+	results, err := api.CheckMachines(
+		params.ModelArgs{ModelTag: st.ModelTag().String()})
+	c.Assert(err, gc.ErrorMatches, "kablooie")
+	c.Assert(results, gc.DeepEquals, params.ErrorResults{})
+}
+
+func (s *Suite) TestCheckMachinesSuccess(c *gc.C) {
+	st := s.Factory.MakeModel(c, nil)
+	defer st.Close()
+
+	fact := factory.NewFactory(st)
+	fact.MakeMachine(c, &factory.MachineParams{
+		InstanceId: "eriatarka",
+	})
+	m := fact.MakeMachine(c, &factory.MachineParams{
+		InstanceId: "volta",
+	})
+	c.Assert(m.Id(), gc.Equals, "1")
+
+	env := mockEnviron{
+		Stub: &testing.Stub{},
+		instances: []*mockInstance{
+			{id: "volta"},
+			{id: "eriatarka"},
+		},
+	}
+	api := s.mustNewAPIWithEnviron(c, &env)
+
+	results, err := api.CheckMachines(
+		params.ModelArgs{ModelTag: st.ModelTag().String()})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results, gc.DeepEquals, params.ErrorResults{})
+}
+
+func (s *Suite) TestCheckMachinesHandlesContainers(c *gc.C) {
+	st := s.Factory.MakeModel(c, nil)
+	defer st.Close()
+
+	fact := factory.NewFactory(st)
+	m := fact.MakeMachine(c, &factory.MachineParams{
+		InstanceId: "birds",
+	})
+	fact.MakeMachineNested(c, m.Id(), nil)
+
+	env := mockEnviron{
+		Stub:      &testing.Stub{},
+		instances: []*mockInstance{{id: "birds"}},
+	}
+	api := s.mustNewAPIWithEnviron(c, &env)
+
+	results, err := api.CheckMachines(
+		params.ModelArgs{ModelTag: st.ModelTag().String()})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results, gc.DeepEquals, params.ErrorResults{})
+}
+
+func (s *Suite) newAPI(environFunc stateenvirons.NewEnvironFunc) (*migrationtarget.API, error) {
 	ctx := facadetest.Context{
 		State_:     s.State,
 		Resources_: s.resources,
 		Auth_:      s.authorizer,
 		StatePool_: state.NewStatePool(s.State),
 	}
 	api, err := migrationtarget.NewAPI(ctx, environFunc)
-	return api, &ctx, err
+	s.AddCleanup(func(*gc.C) { ctx.StatePool().Close() })
```

Comment:
I almost have a PR ready which adds `StatePool` to `StateSuite` (which gets cleaned up automatically). Oh well.